File: public/static/templates/go/external/Command.go
forEach: RelationCommandInfo
fileName: {{commandValue.namePascalCase}}Command.go
path: {{boundedContext.name}}/external
---
package external

type {{commandValue.namePascalCase}}Command struct {
{{#commandValue.fieldDescriptors}}
    {{#isKey}}
    {{/isKey}}
    {{namePascalCase}} {{#typeCheck className}} {{/typeCheck}} `json:"{{nameCamelCase}}"`
{{/commandValue.fieldDescriptors}}
}

<function>
window.$HandleBars.registerHelper('except', function (command) {
    if(command._type.endsWith('Command')) {
        return (command.fieldDescriptors && command.fieldDescriptors.length == 0);
    }
    return true;
});

    window.$HandleBars.registerHelper('typeCheck', function (className) {
        if(className.endsWith("String")){
            return "string"
        }
		else if(className.endsWith("Integer")){
			return "int"
		}
		else if(className.endsWith("Float")){
			return "float64"
		}
		else if(className.endsWith("Long")){
			return "int"
		}

    });

</function>

File: public/static/templates/eventuate/src/main/java/com/eventstorming/domain/Command.java
forEach: Command
representativeFor: Command
fileName: {{namePascalCase}}Command.java
path: {{boundedContext.name}}/{{{options.packagePath}}}/domain
except: {{#except fieldDescriptors}}{{/except}}
---
package {{options.package}}.domain;

import javax.persistence.*;
import java.util.List;
import java.util.Date;
import lombok.Data;

@Data
public class {{namePascalCase}}Command {

{{#fieldDescriptors}}
    {{#isKey}}
        @Id
        //@GeneratedValue(strategy=GenerationType.AUTO)
    {{/isKey}}
        private {{className}} {{nameCamelCase}};
{{/fieldDescriptors}}


}

<function>
window.$HandleBars.registerHelper('except', function (fieldDescriptors) {
    return (fieldDescriptors && fieldDescriptors.length == 0);
});
</function>
